packages/uniforms-mui/src/DateField.tsx

```diff
@@ -0,0 +1,64 @@
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import React from 'react';
+import { FieldProps, connectField, filterDOMProps } from 'uniforms';
+
+/* istanbul ignore next */
+const DateConstructor = (typeof global === 'object' ? global : window).Date;
+const dateFormat = (value?: Date) => value && value.toISOString().slice(0, -8);
+const dateParse = (timestamp: number, onChange: DateFieldProps['onChange']) => {
+  const date = new DateConstructor(timestamp);
+  if (date.getFullYear() < 10000) {
+    onChange(date);
+  } else if (isNaN(timestamp)) {
+    onChange(undefined);
+  }
+};
+
+export type DateFieldProps = FieldProps<
+  Date,
+  TextFieldProps,
+  { labelProps?: object }
+>;
+
+function Date({
+  disabled,
+  error,
+  errorMessage,
+  helperText,
+  InputLabelProps,
+  inputRef,
+  label,
+  labelProps,
+  name,
+  onChange,
+  placeholder,
+  readOnly,
+  showInlineError,
+  value,
+  ...props
+}: DateFieldProps) {
+  return (
+    <TextField
+      disabled={disabled}
+      error={!!error}
+      fullWidth
+      helperText={(error && showInlineError && errorMessage) || helperText}
+      label={label}
+      InputLabelProps={{ shrink: true, ...labelProps, ...InputLabelProps }}
+      inputProps={{ readOnly }}
```

Comment:
I think both approaches would make sense. In the case that `readOnly` comes from the schema then I think it makes sense for it to be overwritten with an explicitly specified one. In case `readOnly` is specified both in `inputProps` and directly as a prop to the field I think it makes more sense for the field prop to take precedence so in the reverse order.

Reply:
I'd go for the second option then.